programs/stake/src/stake_state.rs

```diff
@@ -507,26 +508,42 @@ impl Stake {
     pub fn calculate_points_and_credits(
         &self,
         new_vote_state: &VoteState,
-        stake_history: Option<&StakeHistory>,
+        stake_context: Option<(Epoch, &StakeHistory)>,
         inflation_point_calc_tracer: &mut Option<impl FnMut(&InflationPointCalculationEvent)>,
         fix_stake_deactivate: bool,
     ) -> (u128, u64) {
         // if there is no newer credits since observed, return no point
+        let mut stake_at_target_epoch = None;
+        if let Some((target_epoch, stake_history)) = stake_context {
+            let stake =
+                self.delegation
+                    .stake(target_epoch, Some(stake_history), fix_stake_deactivate);
+            stake_at_target_epoch = Some(stake);
+
+            if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
+                inflation_point_calc_tracer(&InflationPointCalculationEvent::Stake(stake));
+            }
+        }
+
         if new_vote_state.credits() <= self.credits_observed {
             return (0, 0);
         }
 
         let mut points = 0;
         let mut new_credits_observed = self.credits_observed;
 
-        for (epoch, final_epoch_credits, initial_epoch_credits) in
```

Comment:
Looking into this further... I believe this change can deny small stakes delegated to validators with small, but non-zero commissions ever getting paid out, as per https://github.com/solana-labs/solana/blob/d9a07e7ed91c9b67c30338b24f48e866cb9bae91/programs/stake/src/stake_state.rs#L611-L616

Reply:
yeah! correct! I intentionally done this way now. My two cent is posted here: https://github.com/solana-labs/solana/pull/13680#discussion_r527156860